### PR TITLE
added testing files

### PR DESCRIPTION
**Pull request by ASharma2001**
Added Puppeteer and Jest to the project.

### CODE REVIEW

File: __Tests__/website.test.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const sum = require('../source/scripts/sum');
+
+test('adds 1 + 2 to equal 3', () => {
```
Comment (kjl001) [inline __Tests__/website.test.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
test("adds 1 + 2 to equal 3", () => {
```

File: __Tests__/e2e.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const puppeteer = require('puppeteer');
+const URL = "http://127.0.0.1:5501/source/index.html";
+
+describe('basic user flow of the website', () => {
+    // Checking for 6 category cards on the website
+    it('Initial rendering of the page -- check the category cards', async () => {
```
Comment (kjl001) [inline __Tests__/e2e.test.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
    it("Initial rendering of the page -- check the category cards", async () => {
```

File: __Tests__/website.test.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const sum = require('../source/scripts/sum');
```
Comment (kjl001) [inline __Tests__/website.test.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
const sum = require("../source/scripts/sum");
```

File: __Tests__/e2e.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const puppeteer = require('puppeteer');
+const URL = "http://127.0.0.1:5501/source/index.html";
+
+describe('basic user flow of the website', () => {
```
Comment (kjl001) [inline __Tests__/e2e.test.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
describe("basic user flow of the website", () => {
```

File: __Tests__/e2e.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const puppeteer = require('puppeteer');
```
Comment (kjl001) [inline __Tests__/e2e.test.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
const puppeteer = require("puppeteer");
```

File: __Tests__/e2e.test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('basic user flow of the website', () => {
+    // Checking for 6 category cards on the website
+    it('Initial rendering of the page -- check the category cards', async () => {
+        const browser = await puppeteer.launch();
+        const page = await browser.newPage();
+        await page.goto(URL);
+      
+        // Get the all the category cards on the website
+        const numCategoryCards = await page.$$eval('category-card', (categoryCards) => {
```
Comment (kjl001) [inline __Tests__/e2e.test.js:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422468)
```suggestion
        const numCategoryCards = await page.$$eval("category-card", (categoryCards) => {
```

**Comment by Dsuzuki126** [comment]
I think its good, thanks for setting up the test format

**Comment by humanmanwhoisnotdead** [comment]
Yeah, it looks fine for the most part, the only problem I found was the lack of semi-colon on e2e.test.js. Which I don't think is worth preventing the pull request over. Maybe just fix it the next time you do add some tests.

**Pull request merged**